io/include/pcl/io/grabber.h

```diff
@@ -68,9 +68,21 @@ namespace pcl
         * \param[in] callback: the callback function/method
         * \return Connection object, that can be used to disconnect the callback method from the signal again.
         */
-      template<typename T> boost::signals2::connection 
+      template<typename T> boost::signals2::connection
       registerCallback (const std::function<T>& callback);
 
+      /** \brief registers a callback function/method to a signal with the corresponding signature
+        * \param[in] callback: the callback function/method
+        * \return Connection object, that can be used to disconnect the callback method from the signal again.
+        */
+      template<typename T, template<typename> class FunctionT>
+      [[deprecated ("This method will be deprecated in a future version. Please assign the callback to a std::function.")]]
```

Comment:
Why "will be"? `__deprecated__` attribute implies that the thing is already deprecated. Also, the warning message generated from this attribute will automatically contain the name of the entity and a statement that it is deprecated. So our message does not need to say that, it's sufficient to give a suggestion what to use instead, e.g.:

```cpp
[[deprecated ("please assign the callback to a std::function")]] 
```

GCC 7.4 generates the following warning:

```
warning: ‘boost::signals2::connection pcl::Grabber::registerCallback(const FunctionT<T>&) [with T = void(const boost::shared_ptr<const pcl::PointCloud<pcl::PointXYZ> >&); FunctionT = boost::function]’ is deprecated: please assign the callback to a std::function [-Wdeprecated-declarations]
```